app/Model/Project.php

```diff
@@ -1820,4 +1820,42 @@ public function ComputeTestingDayBounds($date)
         $endOfDay = gmdate(FMT_DATETIME, $end_timestamp);
         return [$beginningOfDay, $endOfDay];
     }
+
+
+    /**
+     * Return the testing day (a string in DATETIME format)
+     * for a given date (a date/time string).
+     */
+    public function GetTestingDay($date)
+    {
+        // If the build was started after the nightly start time
+        // then it should appear on the dashboard results for the
+        // subsequent day.
+        $build_datetime = new \DateTime($date);
```

Comment:
@zackgalbreath, is this date in the local CDash server timezone and not UTC?

Reply:
This date (typically a build start time) is actually in the system time zone of the MySQL server. So it could potentially be neither UTC nor the time zone set in `config.local.php`.

> MySQL converts TIMESTAMP values from the current time zone to UTC for storage, and back from UTC to the current time zone for retrieval. [[1]](https://dev.mysql.com/doc/refman/5.7/en/datetime.html)

> When the server starts, it attempts to determine the time zone of the host machine automatically [[2]](https://dev.mysql.com/doc/refman/5.7/en/time-zone-support.html)